dart/constraint/ConstrainedGroup.h

```diff
@@ -107,7 +108,7 @@ class ConstrainedGroup
   std::vector<ConstraintBase*> mConstraints;
 
   ///
-  dynamics::Skeleton* mRootSkeleton;
+  std::shared_ptr<dynamics::Skeleton> mRootSkeleton;
```

Comment:
dynamics::SkeletonPtr is defined in the Skeleton.h header. In places where the Skeleton.h header wouldn't be needed for anything else, I opted to use std::shared_ptr<Skeleton> instead of SkeletonPtr to try to cut down on unnecessary header inclusions.

In theory, it should reduce compilation times that result from changes to the Skeleton.h header. I actually think we might be able to do a lot better in this respect:

We might want to create a SkeletonPtr.h file that just contains the following:

```
#include <memory>

namespace dart {
namespace dynamics {
class Skeleton;
typedef std::shared_ptr<Skeleton> SkeletonPtr;
} // namespace dynamics
} // namespace dart
```

Then we can just include SkeletonPtr.h to access SkeletonPtr, without having a header dependency on Skeleton.h